pkg/cluster/k8sres.go

```diff
@@ -610,6 +735,81 @@ func (c *Cluster) generateStatefulSet(spec *spec.PostgresSpec) (*v1beta1.Statefu
 	return statefulSet, nil
 }
 
+func getEffectiveDockerImage(globalDockerImage, clusterDockerImage string) string {
+	if clusterDockerImage == "" {
+		return globalDockerImage
+	}
+	return clusterDockerImage
+}
+
+func generateScalarSidecarSpec(clusterName, APIKey, serverURL, dockerImage string, containerResources *spec.Resources) (sidecar *spec.Sidecar, present bool) {
+	if APIKey == "" || serverURL == "" || dockerImage == "" {
```

Comment:
does this mean that if this particular sidecar is misconfigured, we silently ignore this fact w/o logging it?